state/presence/presence.go

```diff
@@ -325,6 +351,26 @@ func (w *Watcher) flush() {
 	w.pending = w.pending[:0]
 }
 
+// checkShouldPrune looks at whether we should run a prune step this time
+func (w *Watcher) checkShouldPrune() {
+	chanceToPrune := float64(w.syncsSinceLastPrune) * psuedoRandomFactor
+	if chanceToPrune >= 1.0 || rand.Float64() < chanceToPrune {
+		w.prune()
+	}
+}
+
+// prune cleans out old data in the Beings and Pings tables.
+func (w *Watcher) prune() {
+	logger.Debugf("watcher decided to prune %q and %q", w.beings.Name, w.pings.Name)
+	w.syncsSinceLastPrune = 0
+	pruner := NewBeingPruner(w.modelUUID, w.beings, w.pings, w.delta)
+	err := pruner.Prune()
```

Comment:
at one point the beings were being pruned from a different set of functions than then pings, since they're now in the same object, I'm fine merging it into checkShouldPrune if you feel its better.